README.md

```diff
@@ -49,6 +49,24 @@ The maintainers are unable to guarantee that no significant refactoring will eve
 [similar-work]: https://www.doc.ic.ac.uk/~dorchard/publ/haskell14-effects.pdf
 [ripple]: https://ripple.com/
 
+## Pre-commit
+This repository use [pre-commit](https://pre-commit.com/) to install git commit hooks which will run checks on the repository (currently just clang-format).  To install the git commit hooks locally so they run on each commit you make just set up a virtual environment, install the requirement and run pre-commit install:
+```
+# Set up a virtual environment to install pre-commit
+python3 -m venv .venv
+source ./.venv/bin/activate
+pip install -r ./requirements.txt
```

Comment:
I'd rather do without `requirements.txt` for the time being, just `pip install pre-commit`

Reply:
Done